projects/make-a-vis/src/lib/data-view/shared/data.service.ts

```diff
@@ -50,18 +71,27 @@ export class DataService {
             dataSource.hiddenData = false;
             dataSource.numRows = 0;
             dataSource.streamId = get(recordSet.defaultRecordStream, 'id');
+
             const operator = this.getDataMappingOperator(recordSet.dataVariables, project.graphicVariables, recordSet.id);
             if (recordSet.defaultRecordStream) {
-              recordSet.defaultRecordStream.asObservable().subscribe((changeSet: RawChangeSet<any>) => {
-                dataSource.data = (changeSet.insert || []).map(operator.getter);
-                dataSource.numRows += (changeSet.insert.length || 0) - (changeSet.remove.length || 0);
-                if (changeSet.insert.length > 1) {
-                  dataSource.label = recordSet.labelPlural;
-                }
-              });
+              const data: any[] = [];
+              dataSource.data = recordSet.defaultRecordStream.asObservable().pipe(
+                map((changeSet: RawChangeSet<any>) => {
+                  dataSource.numRows += (changeSet.insert.length || 0) - (changeSet.remove.length || 0);
```

Comment:
Rather then using changeSet, use the cached data to have a more accurate numRows and label